syntax = "proto3";

package todo;

service TodoService {
  rpc CreateTask (CreateTaskRequest) returns (TaskResponse);
  rpc ReadAllTasks (Empty) returns (TasksResponse);
  rpc ReadTodayTasks (Empty) returns (TasksResponse);
  rpc UpdateTask (UpdateTaskRequest) returns (TaskResponse);
  rpc MarkTaskDone (MarkTaskDoneRequest) returns (TaskResponse);
  rpc DeleteTask (DeleteTaskRequest) returns (Empty);
}

message CreateTaskRequest {
  string title = 1;
  string description = 2;
  string due_date = 3; // format: YYYY-MM-DD
}

message UpdateTaskRequest {
  int32 id = 1;
  string title = 2;
  string description = 3;
  string due_date = 4;
}

message MarkTaskDoneRequest {
  int32 id = 1;
}

message DeleteTaskRequest {
  int32 id = 1;
}

message TaskResponse {
  int32 id = 1;
  string title = 2;
  string description = 3;
  string due_date = 4;
  bool done = 5;
}

message TasksResponse {
  repeated TaskResponse tasks = 1;
}

message Empty {}
